ui/src/containers/ConnectionLostDialog.jsx

```diff
@@ -1,88 +1,78 @@
-import React from 'react';
-import { connect } from 'react-redux';
-import { bindActionCreators } from 'redux';
-import { Modal, Button } from 'react-bootstrap';
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Modal, Button, Spinner } from 'react-bootstrap';
 import { showConnectionLostDialog } from '../actions/general';
 
-export class ConnectionLostDialog extends React.Component {
-  constructor(props) {
-    super(props);
-    this.accept = this.accept.bind(this);
-    this.reject = this.reject.bind(this);
-    this.intervalCb = this.intervalCb.bind(this);
-    this.seconds = 60;
-    this.timerid = undefined;
-  }
+import { serverIO } from '../serverIO';
 
-  intervalCb() {
-    this.seconds--;
+export function ConnectionLostDialog() {
+  const dispatch = useDispatch();
 
-    if (this.refs.countdown) {
-      this.refs.countdown.innerHTML = this.seconds;
-    }
+  const [timerElapsed, setTimerElapsed] = useState(30);
 
-    if (this.seconds < 1) {
-      this.seconds = 0;
-      this.accept();
-    }
-  }
+  const [buttonLoading, setbuttonLoading] = useState(false);
 
-  accept() {
-    this.forceUpdate();
-  }
+  const show = useSelector((state) => state.general.showConnectionLostDialog);
 
-  reject() {
-    this.props.hide();
+  function accept() {
+    // just a visual effect
+    setbuttonLoading(true);
+    setTimeout(() => setbuttonLoading(false), 800);
+    // A force update action ?
+    window.location.reload(true);
   }
 
-  render() {
-    this.seconds = 60;
+  function intervalCb() {
+    setTimerElapsed(timerElapsed - 1);
 
-    if (this.timerid) {
-      clearInterval(this.timerid);
+    if (timerElapsed === 1) {
+      accept();
+      setTimeout(() => setTimerElapsed(30), 1000);
     }
+  }
 
-    this.timerid = setInterval(this.intervalCb, 1000);
-    setTimeout(() => {
-      clearInterval(this.timerid);
-    }, this.seconds * 1000);
+  useEffect(() => {
+    if (show) {
+      const forceUpdateInterval = setInterval(intervalCb, 1000);
+      return () => clearInterval(forceUpdateInterval);
+    }
+    return null;
+  });
 
-    return (
-      <Modal show={this.props.show} onHide={this.props.hide}>
-        <Modal.Header>
-          <Modal.Title>Connection lost</Modal.Title>
-        </Modal.Header>
-        <Modal.Body>
-          Ooops ! There seems to be a problem with the internet connection, the
-          connection to the server was lost. Trying to reconnect in{' '}
-          <span ref="countdown">{this.seconds}</span> s
-        </Modal.Body>
-        <Modal.Footer>
-          <Button onClick={this.accept}> Try again </Button>
-        </Modal.Footer>
-      </Modal>
-    );
+  function rejectHide() {
+    // if server still disconnect after 9 second show dialog again
+    if (serverIO.isServerDisconnect()) {
+      setTimeout(() => {
+        dispatch(showConnectionLostDialog(true));
+        setTimerElapsed(30);
+      }, 9000);
```

Comment:
Perhaps in `serverIO`, you could start a `setInterval` on line 126 (around the `dispatch` call of the `showConnectionLostDialog` action)? This would maintain the separation of concerns between `serverIO` and the connection lost dialog (which should only care about the Redux state `state.general.showConnectionLostDialog`)

> Don't forget to clean up the `setInterval` the same way as the `setTimeout`.

That being said, this feels like a hack to account for the fact that the dialog is dismissable...

Reply:
Yes, I don't really know why we added this countdown timer. I also think a "Try again" button as @axelboc mentions above is nicer solution and makes things a bit simpler.

It would also be good if we could prevent the dialog from being closed as well.

Reply:
I guess the countdown was added to make the automatic page reload more intuitive for the user. This was initially implemented to handle server  and user  state synchronization   

Previously, remote users who left the screen open and returned later might see the queue still running, while the local contact reported it as finished. The auto force update with a countdown helped prevent this issue by ensuring synchronization after a timeout.

this is handle differently now.


 it seems showConnectionLostDialog is not so relevant now.

Reply:
Not entirely certain that was why. I think we just tried to be informative, and it ended up to be misleading in stead. Anyways the suggested solution is nicer

Reply:
I guess showing the websocket reconnection timer would perhaps be more useful ... but definitely not worth the effort and complexity IMO :joy:

Reply:
yes, that would be more correct. But a clear message saying that it will automatically reconnect is also fine :)